test/rubygems/test_gem_platform.rb

```diff
@@ -143,6 +145,7 @@ def test_initialize
     test_cases.each do |arch, expected|
       platform = Gem::Platform.new arch
       assert_equal expected, platform.to_a, arch.inspect
+      assert_equal expected, Gem::Platform.new(platform.to_s).to_a, arch.inspect
```

Comment:
An alternative test might be `assert_equal platform, Gem::Platform.new(platform.to_s), arch.inspect`